R/git.R

```diff
@@ -0,0 +1,84 @@
+#' Check if git repo is clean
+#'
+#' Check that git repo is up to date with remote by checking
+#' \code{git status --porcelian}. If import_dir is not a git root directory
+#' this will search parents of the import_dir for the git root
+#'
+#' @param import_dir The directory of the dettl import to check.
+#' @return True if repo is up to date. False otherwise.
+#'
+#' @keywords internal
+git_is_clean <- function(import_dir) {
+  root_dir <- git_root_directory(import_dir)
+  git_status(root_dir)$clean
+}
+
+#' Check git status
+#'
+#' Check that git status using \code{git status --porcelian}. Optionally
+#' ignoring untracked files.
+#'
+#' @param root The git project root directory.
+#' @param ignore_untracked Whether untracked files should be ignored or not.
+#' Passed as \code{--untracked-files} arg to \code{git status}
+#' @return Result from git including output and indication of success.
+#'
+#' @keywords internal
+git_status <- function(root = NULL, ignore_untracked = FALSE) {
+  args <- c("status", "--porcelain",
+            if (ignore_untracked) "--untracked-files=no")
```

Comment:
Do you have plans for this option? It's not used or tested anywhere I can see

Reply:
No plans for this, it was just copied over from orderly. Not used anywhere or tested. I just left for the case that we may want it in the future. But I guess that probably confuses things. I've taken it out.